File: Sum Root to Leaf Numbers.java
/**
 * Definition for binary tree
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
public class Solution {
    int sum;
    public int sumNumbers(TreeNode root) {
        // Start typing your Java solution below
        // DO NOT write main() function
        sum = 0;
        if(root!=null)
        {
            findLeefNode(root, 0);
        }
        return sum;
        
    }
    
    private void findLeefNode(TreeNode node, int val)
    {
        val = 10*val + node.val;
        if(node.left==null && node.right==null)
        {
            sum += val;
        }
        else
        {
            if(node.left!=null)
            {
                findLeefNode(node.left, val);
            }
            if(node.right!=null)
            {
                findLeefNode(node.right, val);
            }
        }
    }
}

File: Sum Root to Leaf Numbers.cpp
/**
 * Definition for binary tree
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */
class Solution {
public:
   
    int sumNumbers(TreeNode *root) {
        // Start typing your C/C++ solution below
        // DO NOT write int main() function
        int num = 0;
        vector<int> result;
        findSum(root, num, result);
        int sum = 0;
        for (int i = 0; i< result.size(); i++)
        {
            sum += result[i];
        }
        return sum;
    }
    
    void findSum(TreeNode*root, int num, vector<int> &result)
    {
        if(root==NULL)
        {
            return;
        }
        else
        {
            int newnum = num*10 + root->val;
            if(root->left==NULL&&root->right==NULL)
                result.push_back(newnum);
            findSum(root->left, newnum, result);
            findSum(root->right, newnum, result);
        }
    }
    
};